js/woo.js

```diff
@@ -0,0 +1,984 @@
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchMarkets', undefined, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PublicGetInfo',
+        });
+        let data = undefined;
+        if (marketType === 'spot') {
+            const response = await this[method] (query);
+            //
+            // {
+            //     rows: [
+            //         {
+            //             symbol: "SPOT_AAVE_USDT",
+            //             quote_min: 0,
+            //             quote_max: 100000,
+            //             quote_tick: 0.01,
+            //             base_min: 0.01,
+            //             base_max: 7284,
+            //             base_tick: 0.0001,
+            //             min_notional: 10,
+            //             price_range: 0.1,
+            //             created_time: "0",
+            //             updated_time: "1639107647.988",
+            //             is_stable: 0
+            //         },
+            //         ...
+            //     success: true
+            // }
+            //
+            data = this.safeValue (response, 'rows', []);
+        }
+        const result = [];
+        for (let i = 0; i < data.length; i++) {
+            const market = data[i];
+            const marketId = this.safeString (market, 'symbol');
+            const parts = marketId.split ('_');
+            const baseId = parts[1];
+            const quoteId = parts[2];
+            const marketTypeVal = parts[0].toLowerCase ();
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const minQuote = this.safeNumber (market, 'quote_min');
+            const maxQuote = this.safeNumber (market, 'quote_max');
+            const minBase = this.safeNumber (market, 'base_min');
+            const maxBase = this.safeNumber (market, 'base_max');
+            const priceScale = this.safeNumber (market, 'quote_tick');
+            const quantityScale = this.safeNumber (market, 'base_tick');
+            const minCost = this.safeNumber (market, 'min_notional');
+            result.push ({
+                'id': marketId,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'settle': undefined,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'settleId': undefined,
+                'type': marketTypeVal,
+                'spot': true,
+                'margin': false,
+                'swap': false,
+                'future': false,
+                'option': false,
+                'active': undefined,
+                'contract': false,
+                'linear': undefined,
+                'inverse': undefined,
+                // Fee is in %, so divide by 100
+                'taker': undefined,
```

Comment:
Let's remove `taker` and `maker` from markets if they're not defined in this endpoint, they will be filled in later in the base code here:

- https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L724

Reply:
This is an exceptional case, cause otherwise the undefined values will overwrite the fees inherited from the exchange-wide .fees property.